Userland/Libraries/LibSoftGPU/Device.cpp

```diff
@@ -1258,33 +1257,33 @@ void Device::set_light_model_params(const LightModelParameters& lighting_model)
     m_lighting_model = lighting_model;
 }
 
-Gfx::RGBA32 Device::get_backbuffer_pixel(int x, int y)
+ColorType Device::get_color_buffer_pixel(int x, int y)
 {
     // FIXME: Reading individual pixels is very slow, rewrite this to transfer whole blocks
-    if (x < 0 || y < 0 || x >= m_render_target->width() || y >= m_render_target->height())
+    if (x < 0 || y < 0 || x >= m_frame_buffer->rect().width() || y >= m_frame_buffer->rect().height())
         return 0;
 
-    return m_render_target->scanline(y)[x];
+    return m_frame_buffer->color_buffer()->scanline(y)[x];
 }
 
-float Device::get_depthbuffer_value(int x, int y)
+DepthType Device::get_depthbuffer_value(int x, int y)
 {
     // FIXME: Reading individual pixels is very slow, rewrite this to transfer whole blocks
-    if (x < 0 || y < 0 || x >= m_render_target->width() || y >= m_render_target->height())
+    if (x < 0 || y < 0 || x >= m_frame_buffer->rect().width() || y >= m_frame_buffer->rect().height())
         return 1.0f;
 
-    return m_depth_buffer->scanline(y)[x];
+    return m_frame_buffer->depth_buffer()->scanline(y)[x];
 }
 
-NonnullRefPtr<Image> Device::create_image(ImageFormat format, unsigned width, unsigned height, unsigned depth, unsigned levels, unsigned layers)
+NonnullRefPtr<Image> Device::create_image(unsigned width, unsigned height, unsigned depth, unsigned levels, unsigned layers)
```

Comment:
Seeing as this interface will also be used for other drivers we should not remove the format parameter but simply verify here that we get passed a supported format. In the future we should return a list of supported image formats via the DeviceInfo struct.

Reply:
Agreed, partially restored this so we always provide `BGRA8888`.